www/apache2/Portfile

```diff
@@ -211,7 +211,7 @@ variant eventmpm conflicts preforkmpm workermpm description {Use event MPM (expe
 }
 
 if {![variant_isset workermpm] && ![variant_isset eventmpm]} {
-    default_variants +preforkmpm
+    default_variants +preforkmpm +mpms_shared_all
```

Comment:
This will not make +mpms_shared_all a default variant if the user selected +workermpm or +eventmpm. Is that what you intended?

Is there any reason for the mpms_shared_all variant to exist at all -- is there any reason why a user would want to disable that functionality? Why not build it all the time?

(This is the first time I've noticed that this variant was added to the port and I wasn't aware of this apache functionality before.)

If you add a new default variant, you must increase the revision. See https://guide.macports.org/#reference.keywords

Reply:
> This will not make +mpms_shared_all a default variant if the user selected +workermpm or +eventmpm. Is that what you intended?

Ah, ok. I'll remove it then.

> Is there any reason for the mpms_shared_all variant to exist at all -- is there any reason why a user would want to disable that functionality? Why not build it all the time

We added it here: #3384. It's a really great function and you don't need to rebuild Apache when you want to change mpm, + you can use all mpms at the same time (i diff vhost though).

Yes, if that's ok, I'd much more prefer that as a default arg. I guess it wouldn't be needed to req any variant for the apache2handler subport since it's already been built, put perhaps not enabled. Just need a note for that, to make sure they use it.

Reply:
Actually, with `mpms_shared_all` as a default arg in the build, one _could_ also add prefork and then remove all the mpm variants. It'll just be a matter of configuring for the user.

```
--enable-mpms-shared=all \
--with-mpm=prefork \
```

…will build all 3, and add all 3 to httpd.conf, with the prefork version uncommented.